File: StudentPortal_backend/src/main/java/service/NotificationService.java
package service;

import org.springframework.stereotype.Service;

import model.Notification;
import repository.NotificationRepository;

import java.util.List;

@Service
public class NotificationService {

    private final NotificationRepository repository;

    public NotificationService(NotificationRepository repository) {
        this.repository = repository;
    }

    public Notification createNotification(Notification notification) {
        return repository.save(notification);
    }

    public List<Notification> getUserNotifications(Long userId) {
        return repository.findByUserId(userId);
    }

    public List<Notification> getUnreadNotifications(Long userId) {
        return repository.findByUserIdAndIsRead(userId, false);
    }

    public long getUnreadCount(Long userId) {
        return repository.countByUserIdAndIsRead(userId, false);
    }

    public Notification markAsRead(Long notificationId) {
        Notification notification = repository.findById(notificationId).orElse(null);
        if (notification != null) {
            notification.setRead(true);
            return repository.save(notification);
        }
        return null;
    }

    public void markAllAsRead(Long userId) {
        List<Notification> notifications = repository.findByUserIdAndIsRead(userId, false);
        notifications.forEach(n -> n.setRead(true));
        repository.saveAll(notifications);
    }

    public void deleteNotification(Long id) {
        repository.deleteById(id);
    }
}

File: student-portal-frontend/src/services/NotificationService.js
import axios from "axios";

const BASE_URL = "http://localhost:8080/api/notifications";

export const createNotification = (notification) => {
  return axios.post(BASE_URL, notification);
};

export const getUserNotifications = (userId) => {
  return axios.get(`${BASE_URL}/user/${userId}`);
};

export const getUnreadNotifications = (userId) => {
  return axios.get(`${BASE_URL}/unread/${userId}`);
};

export const getUnreadCount = (userId) => {
  return axios.get(`${BASE_URL}/unread-count/${userId}`);
};

export const markAsRead = (notificationId) => {
  return axios.put(`${BASE_URL}/mark-read/${notificationId}`);
};

export const markAllAsRead = (userId) => {
  return axios.put(`${BASE_URL}/mark-all-read/${userId}`);
};

export const deleteNotification = (id) => {
  return axios.delete(`${BASE_URL}/${id}`);
};